content/articles/testing-for-broken-authentication-in-web-apps/index.md

```diff
@@ -0,0 +1,117 @@
+
+This article seeks to demonstrate how an attacker tests for broken authentication in a web application and how to prevent them.
+
+> **Disclaimer:** This article is for educational purposes only.
+
+### Testing for Broken Authentication
+To carry out this attack, I'll be making use of an intentionally vulnerable web application called [Juice shop](https://github.com/juice-shop/juice-shop).
+
+![index page](/engineering-education/testing-for-broken-authentication-in-web-apps/index.png)
+
+After starting up the web application, I'll proceed to the login page for testing.
+
+![login page](/engineering-education/testing-for-broken-authentication-in-web-apps/login.png)
+
+Firstly, I want to check how the application reacts if I provide incorrect login credentials.
+
+![login error](/engineering-education/testing-for-broken-authentication-in-web-apps/error.png)
+
+From the above screenshot, an error message is displayed. Some applications simply come out to tell you the particular field that contains the wrong credentials. something like `incorrect password` where the username provided is correct, or `invalid username` where the password is correct but doesn't match the username provided.
+
+This is a poor practice because it allows for user enumeration. When an attacker uses brute force to guess valid users in a system, this is known as user enumeration.
+
+The next thing I want to do is click on the `Forgot Password?` option. which requires I fill out a form.
+
+![password reset](/engineering-education/testing-for-broken-authentication-in-web-apps/password-reset.png)
+
+Again, I provided an incorrect email and noticed I was unable to click or fill other parts of the form. Interesting!
+
+What if I provided an email address I think is valid, will I be able to complete the form?
+
+![password reset](/engineering-education/testing-for-broken-authentication-in-web-apps/question.png)
+
+After providing a valid email address, the form allows me to click and fill in other fields. Again, this allows for user enumeration. 
+
+Now that the application just confirmed the validity of the email address I provided, the next field is to fill the security question `Mother's maiden name?`. With the use of [open source Intelligence](https://www.recordedfuture.com/open-source-intelligence-definition/) an attacker can easily figure out the answer to this question. In some cases, it just might be the user's last name. Then the attacker moves on to change the password of that user.
+
+This is just one scenario that shows the impact of broken authentication on a login form. Another scenario is weak passwords and default credentials.
+
+Some web applications still allow the use of weak and well-known or default credentials such as `admin: admin` `password1234`. Applications that do not enforce a strong password policy are highly susceptible to this attack.
+
+To demonstrate this, I'll try to login into the admin account and intercept the request using the burp suite.
```

Comment:
Can you brief more about burp suite usage? How to use and what to do?

Add URL for it too.

Reply:
What does it do? How does it help here? - Explain